libkbfs/folder_branch_ops.go

```diff
@@ -1889,17 +1928,22 @@ func (fbo *folderBranchOps) finalizeMDWriteLocked(ctx context.Context,
 		return err
 	}
 
+	mdID, err := fbo.config.Crypto().MakeMdID(&md.BareRootMetadata)
+	if err != nil {
+		return err
+	}
+
 	fbo.headLock.Lock(lState)
 	defer fbo.headLock.Unlock(lState)
-	err = fbo.setHeadSuccessorLocked(ctx, lState, md)
+	err = fbo.setHeadSuccessorLocked(ctx, lState, MakeImmutableRootMetadata(md, mdID))
```

Comment:
Maybe better to make the immutable MD a variable, and use it's const object in the calls below.